### ISSUE TITLE
Email confirmation not sending using non-proxy registration

### ISSUE THREAD

**Issue by ikkysleepy** [apiato/apiato#658]
### Description:
When Creating a new user using the  private user registration api url the end-user does not receive an confirmation email.

User - Register User (create client) - URL:
http://api.apiato.test/v1/register

The following config is set to true:
```require_email_confirmation```

### Expected Behavior:
2 emails
 - 1 welcome  email - working
 -  1 verification confirmation email / error because  I have no template ?

### Additional Context:
The nodejs Web API is on on container on the same Repo / Host . Mail is configured to use https://mailtrap.io/ . I am able to login using the  OAuth 2.0 - A For first-party clients using all the required fields. 

The documentation says the following:
```When email confirmation is enabled (value set to true), the API throws an exception, if the User is not yet confirmed.```
But when I login, using http://api.apiato.test/v1/oauth/token ,  I do not get an API exception error, even though this is set to true.

### Versions:
- Apiato Version: 10.0.
- PHP Version: 7.4.24

### Steps To Reproduce:
1. Set ```require_email_confirmation``` to true
2. Register a user using ```http://api.apiato.test/v1/register```

**Comment by Mohammad-Alavi** [comment]
The email confirmation in v10 is almost none-existence and not implemented correctly. I found this out after releasing v10. I fully implemented and tested email confirmation process for the next version (v11) but it was a hug work to implement it for v10 too and i didn't have that much time.

TL;DR Email confirmation is not implemented in v10 and you have to implement it youself. You can see the master branch to get some ideas about how to do it.
